logverification/app/appentry.go

```diff
@@ -208,51 +187,140 @@ func (ae *AppEntry) LogTenant() (string, error) {
 
 }
 
-// ExtraBytes gets the extrabytes of the corresponding log entry.
-func (ae *AppEntry) ExtraBytes() []byte {
-	return ae.extraBytes
+// TrieEntry gets the corresponding log trie entry for the app entry.
+func (ae *AppEntry) TrieEntry(options ...MassifGetterOption) ([]byte, error) {
+
+	massifContext, err := ae.Massif(options...)
+	if err != nil {
+		return nil, err
+	}
+
+	trieEntry, err := massifContext.GetTrieEntry(ae.MMRIndex())
+	if err != nil {
+		return nil, err
+	}
+
+	return trieEntry, nil
+
 }
 
-// SerializedBytes gets the serialized bytes used to derive the mmr entry.
-func (ae *AppEntry) SerializedBytes() []byte {
-	return ae.mmrEntryFields.serializedBytes
+// ExtraBytes gets the extrabytes of the corresponding log entry.
+func (ae *AppEntry) ExtraBytes(options ...MassifGetterOption) ([]byte, error) {
+
+	trieEntry, err := ae.TrieEntry(options...)
+	if err != nil {
+		return nil, err
+	}
+
+	return massifs.GetExtraBytes(trieEntry, 0, 0), nil
 }
 
-// Domain gets the domain byte used to derive the mmr entry.
-func (ae *AppEntry) Domain() byte {
-	return ae.mmrEntryFields.domain
+// IDTimestamp gets the idtimestamp of the corresponding log entry.
+func (ae *AppEntry) IDTimestamp(options ...MassifGetterOption) ([]byte, error) {
+
+	trieEntry, err := ae.TrieEntry(options...)
+	if err != nil {
+		return nil, err
+	}
+
+	return massifs.GetIdtimestamp(trieEntry, 0, 0), nil
 }
 
 // MMRSalt derives the MMR Salt of the corresponding log entry from the app data.
 // MMRSalt is the datatrails provided fields included on the MMR Entry.
 //
 // this is (extrabytes | idtimestamp) for any apps that adhere to log entry version 1.
-func (ae *AppEntry) MMRSalt() ([]byte, error) {
+func (ae *AppEntry) MMRSalt(options ...MassifGetterOption) ([]byte, error) {
 
 	mmrSalt := make([]byte, MMRSaltSize)
 
-	copy(mmrSalt[:ExtraBytesSize], ae.extraBytes)
+	extraBytes, err := ae.ExtraBytes(options...)
+	if err != nil {
+		return nil, err
+	}
 
-	// get the byte representation of idtimestamp
-	idTimestamp, _, err := massifs.SplitIDTimestampHex(ae.merkleLogCommit.Idtimestamp)
+	idTimestamp, err := ae.IDTimestamp(options...)
 	if err != nil {
 		return nil, err
 	}
 
-	idTimestampBytes := make([]byte, IDTimestapSizeBytes)
-	binary.BigEndian.PutUint64(idTimestampBytes, idTimestamp)
+	copy(mmrSalt[:ExtraBytesSize], extraBytes)
 
-	copy(mmrSalt[ExtraBytesSize:], idTimestampBytes)
+	copy(mmrSalt[ExtraBytesSize:], idTimestamp)
 
 	return mmrSalt, nil
 }
 
+// MMREntry derives the mmr entry of the corresponding log entry from the app data.
+//
+// MMREntry is:
+//   - H( Domain | MMR Salt | Serialized Bytes)
+//
+// The MMR Salt is sourced from the corresponding log entry
+func (ae *AppEntry) MMREntry(options ...MassifGetterOption) ([]byte, error) {
```

Comment:
so with `MassifGetterOption` is there any reason those would change within the lifetime of an AppEntry?

maybe we should just add them to `NewAppEntry` ?

Reply:
Moved to slack thread :)

Reply:
resolved by keeping the massif options within the function, but moving to just a massif context.